docs/promoting-artifacts.md

```diff
@@ -0,0 +1,117 @@
+
+1. Push and sign the `net-monitor:v1` container image to the **Wabbit Networks** public registry
+2. Validate the `net-monitor:v1` image meets the acceptance criteria of ACME Rockets
+3. Promote the `net-monitor:v1` image to the ACME Rockets registry
+4. Sign the `net-monitor:v1` image with the **ACME Rockets library** key, indicating its valid for internal consumption
+
+![](../media/notary-e2e-scenarios.svg)
+
+## Getting Started
+- Setup a few environment variables.  
+  >Note see [Simulating a Registry DNS Name](#simulating-a-registry-dns-name) to use `registry.wabbit-networks.io`
+  ```bash
+  export PUBLIC_PORT=5000
+  export PUBLIC_REGISTRY=localhost:${PUBLIC_PORT}
+  export PUBLIC_REPO=${PUBLIC_REGISTRY}/net-monitor
+  export PUBLIC_IMAGE=${PUBLIC_REPO}:v1
+
+  export PRIVATE_PORT=5050
+  export PRIVATE_REGISTRY=localhost:${PRIVATE_PORT}
+  export PRIVATE_REPO=${PRIVATE_REGISTRY}/net-monitor
+  export PRIVATE_IMAGE=${PRIVATE_REPO}:v1
+  ```
+- Install [Docker Desktop](https://www.docker.com/products/docker-desktop) for local docker operations
+- Run a local registry representing the Wabbit Networks **public** registry
+  ```bash
+  docker run -d -p ${PUBLIC_PORT}:5000 ghcr.io/oras-project/registry:latest
+  ```
+- Run a local registry representing the ACME Rockets **private** registry
+  ```bash
+  docker run -d -p ${PRIVATE_PORT}:5000 ghcr.io/oras-project/registry:latest
+  ```
+## Building and Pushing the Public Image
+
+- Build and push the `net-monitor` software
+  ```bash
+  docker build -t $PUBLIC_IMAGE https://github.com/wabbit-networks/net-monitor.git#main
+
+  docker push $PUBLIC_IMAGE
+  ```
+- Generate a self-signed test certificate for signing artifacts
+  The following will generate a self-signed X.509 certificate under the `~/config/notation/` directory
+  ```bash
+  notation cert generate-test "wabbit-networks.io"
+  ```
+- Sign the container image
+  ```bash
+  notation sign $PUBLIC_IMAGE
+  ```
+- List the image, and any associated signatures
+  ```bash
+  notation list $PUBLIC_IMAGE
+  ```
+
+## Import the Public Image
+
+- Validate the image is signed with a key that fits within the ACME Rockets policy
+  ```bash
+  notation verify $PUBLIC_IMAGE
+  ```
+- The above command should fail, as the Wabbit Networks public key has not yet been configured
+- Configure the Wabbit Networks key for validation, and re-validate
+  ```bash
+  notation cert add -n "wabbit-networks.io" ~/.config/notation/certificate/wabbit-networks.io.crt
+  notation verify $PUBLIC_IMAGE
+  ``` 
+- Create a test cert for the ACME Rockets Library key
+  ```bash
+  notation cert generate-test "acme-rockets.io-library"
+  ```
+- To support tag update scenarios, the image must be signed with a new signature and pushed with all artifact references
+- Sign the imported image, locally
+  ```bash
+  docker pull $PUBLIC_IMAGE
+  docker tag $PUBLIC_IMAGE $PRIVATE_IMAGE
+  docker notation sign $PRIVATE_IMAGE
```

Comment:
does this notation sign command have to be preceded by "docker".  Is the assumption here that a notation plugin for Docker is present?